modules/data-entry/src/main/frontend/src/resourceQuery/ResourceQuery.jsx

```diff
@@ -0,0 +1,525 @@
+  let registerInfoButton = (id, node) => {
+    // List items getting deleted will overwrite new browser button refs, so
+    // we must ignore deregistration events
+    if (node) {
+      buttonRefs[id] = node;
+    }
+  }
+
+  // Event handler for clicking away from the info window while it is open
+  let closeInfo = (event) => {
+    setResourcePath("");
+  }
+
+  let closeSuggestions = () => {
+    if (clearOnClick && anchorEl?.current) {
+      anchorEl.current.value = "";
+    }
+    if (focusAfterSelecting) {
+      anchorEl?.current?.select();
+    }
+    setSuggestionsVisible(false);
+  }
+
+  let updateSelection = (selectedEntries, removedEntries) => {
+    selectedEntries && selectedEntries.map(item => onClick(item[VALUE_POS], item[LABEL_POS]));
+    removedEntries && removedEntries.map(item => onRemoveOption(item[VALUE_POS], item[LABEL_POS]));
+
+    // Set input value to selected term label or initial selection label if single answer question
+    if (maxAnswers === 1) {
+      if (selectedEntries?.length > 0) {
+        // Search in default answer options
+        !isDefaultOption(selectedEntries[0][VALUE_POS]) && setInputValue(selectedEntries[0][LABEL_POS]);
+      }
+    }
+    if (selectedEntries || removedEntries) {
+      setSuggestionsVisible(false);
+      setResourcePath("");
+      maxAnswers != 1 && setInputValue("");
+    } else {
+      anchorEl.current.focus();
+    }
+  }
+
+  if (disabled && anchorEl?.current) {
+    // Alter our text to either the override ("Please select at most X options")
+    // or empty it
+    anchorEl.current.value = "";
+    anchorEl.current.blur();
+  }
+
+  return (
+      <div>
+        {props.children}
+
+        <div className={variant == "labeled" ? classes.searchWrapper : ""}>
+          {variant == "labeled" ?
+          <FormControl className={isNested ? classes.nestedSearchInput : classes.search}>
+            <InputLabel
+              classes={{
+                root: classes.searchLabel,
+                shrink: classes.searchShrink,
+              }}
+            >
+              { /* Cover up a bug that causes the label to overlap the value:
+                   if it has a displayed value and isn't focused, don't show the label
+                 */ }
+              { (document.activeElement === anchorEl.current || (!value && !(anchorEl.current?.value))) ? 'Search' : ''}
+            </InputLabel>
+            {inputEl}
+          </FormControl>
+          :
+          inputEl}
+          <LinearProgress className={classes.progressIndicator + " " + (suggestionsLoading ? "" : classes.inactiveProgress)}/>
+        </div>
+        {/* Suggestions list using Popper */}
+        <Popper
+          open={suggestionsVisible}
+          anchorEl={anchorEl.current}
+          transition
+          className={
+            classNames({ [classes.popperClose]: !open })
+            + " " + classes.popperNav
+            + " " + classes.popperListOnTop
```

Comment:
Could put the entire thing in the classNames call to avoid the repetition of `" " +`:
```suggestion
            classNames({ [classes.popperClose]: !open },
            classes.popperNav,
            classes.popperListOnTop)
```

Reply:
Fixed in [bf9b90e](https://github.com/data-team-uhn/cards/pull/1043/commits/bf9b90e18010034433e18f69faa7e4537ecad1d6) on the `CARDS-1784` branch.